database/service_db.go

```diff
@@ -115,10 +114,12 @@ func (d *LevelDBServiceDB) delete(tx *leveldb.Transaction, hashOrSid string) err
 	if err != nil {
 		return err
 	}
-	if err := tx.Delete([]byte(hashKeyPrefix+s.Hash), nil); err != nil {
-		return err
+	if s.Sid != "" {
```

Comment:
So they are not mandatory right now (they are marked as optional).

Reply:
What about volumes ? 2 services with no sid will share the same volume.
Right now they are not mandatory in the mesg.yml but generated with a value by default (`_HASH` if i remember correctly) so the definition file is optional but in the core mandatory.

This change breaks volumes for services

Reply:
In service manager, if sid is empty it uses the hash.
The behavior is exactly like before.

Reply:
great I didn't see that. Also behavior is not exactly, we are taking the hash, before we where appending an character in front of the hash to make sure that we cannot create a service with the hash of another one as sid and then overwrite the previous service/volume. But anyway that was not a strong security anyway.

I still feel more confident to make sure that we can rely on a service to have a sid (that is generate when we deploy it like the hash) instead of having to check all the time we want to use the sid. I'm not necessary against it either it's just something I would like to have agreement to see the pro and con on that and not change something that impact the data.

Reply:
In that case, we have to generate different sid because one we generated one is an invalid domain. Also, sid should be used only for local development and should not be mandatory.

Reply:
sid is the base for the markeplace too so it's not only about local development. We put sid as optional with generation in order not to break previous services but every services needs to have a sid to identify this service and "group" the different version to the same service.

How do you manage services without sid on the marketplace for example ?